lib/hanami/action/request.rb

```diff
@@ -1,44 +1,58 @@
 require 'rack/request'
+require 'securerandom'
 
 module Hanami
-  module Action
+  class Action
     # An HTTP request based on top of Rack::Request.
     # This guarantees backwards compatibility with with Rack.
     #
     # @since 0.3.1
     #
     # @see http://www.rubydoc.info/gems/rack/Rack/Request
     class Request < ::Rack::Request
-      # @raise [NotImplementedError]
-      #
-      # @since 0.3.1
-      # @api private
-      def content_type
-        raise NotImplementedError, 'Please use Action#content_type'
+      HTTP_ACCEPT = "HTTP_ACCEPT".freeze
+      REQUEST_ID  = "hanami.request_id".freeze
+      DEFAULT_ACCEPT = "*/*".freeze
+      DEFAULT_ID_LENGTH = 16
+
+      attr_reader :params
+
+      def initialize(env, params)
+        super(env)
+        @params = params
       end
 
-      # @raise [NotImplementedError]
-      #
-      # @since 0.3.1
-      # @api private
-      def session
-        raise NotImplementedError, 'Please include Action::Session and use Action#session'
+      def id
+        # FIXME make this number configurable and document the probabilities of clashes
+        @id ||= @env[REQUEST_ID] = SecureRandom.hex(DEFAULT_ID_LENGTH)
```

Comment:
is not better a UUID by default?

Reply:
It's a tradeoff, UUID has nearly no chances to have collisions, but it's slower. That's why we need a fast enough algorithm with reasonable low collision probabilities.

Also, I'd make it somehow configurable (maybe via a lambda), so devs can provide their own strategy if they have custom needs.

Does it makes sense?

Reply:
yes, it's perfect ;) I just wanted to know the reason of use that :)

Reply:
For the record, this is the benchmark: https://gist.github.com/jodosha/a00192eb0b4108c3eab8b542b9f1ad39

UUID is almost 2.4x times slower than HEX 16.